File: python_files/raylib_redo/centered_text.py
import pyray as p

SCREEN_WIDTH: int = 800
SCREEN_HEIGHT: int = 600
SCREEN_TITLE: str = "Centered Text"
SCREEN_BACKGROUND: p.Color = p.RAYWHITE
GAME_FPS: int = 60


def main() -> None:
    p.init_window(SCREEN_WIDTH, SCREEN_HEIGHT, SCREEN_TITLE)
    p.set_target_fps(GAME_FPS)

    width: float = 400.0
    height: float = 200.0
    rect: p.Rectangle = p.Rectangle(
        p.get_screen_width() / 2 - width / 2,
        p.get_screen_height() / 2 - height / 2,
        width,
        height,
    )

    num: int = 0

    while not p.window_should_close():
        if p.is_key_pressed(p.KeyboardKey.KEY_UP):
            num += 1
        if p.is_key_pressed(p.KeyboardKey.KEY_DOWN):
            num -= 1

        p.begin_drawing()
        p.clear_background(SCREEN_BACKGROUND)

        p.draw_rectangle_lines_ex(rect, 5.0, p.BLACK)

        text: str = str(num)
        font_size: float = 40.0
        text_width: float = p.measure_text(text, int(font_size))
        text_x: float = rect.x + rect.width / 2 - text_width / 2
        text_y: float = rect.y + rect.height / 2 - font_size / 2
        spacing: float = 7.0
        tint: p.Color = p.BLACK

        p.draw_text_ex(
            p.get_font_default(),
            text,
            p.Vector2(text_x, text_y),
            font_size,
            spacing,
            tint,
        )

        p.end_drawing()

    p.close_window()


if __name__ == "__main__":
    main()

File: c_files/raylib_redo/centered_text.c
#include <raylib.h>

void
center_and_draw_text(int num, Rectangle rect)
{
  const char* text = TextFormat("%d", num);
  int font_size = 40;
  int text_width = MeasureText(text, font_size);
  int pos_x = rect.x + rect.width / 2 - text_width / 2;
  int pos_y = rect.y + rect.height / 2 - font_size / 2;
  Color color = BLACK;

  DrawText(text, pos_x, pos_y, font_size, color);
}

int
main(void)
{
  const int SCREEN_WIDTH = 800;
  const int SCREEN_HEIGHT = 600;
  const char SCREEN_TITLE[] = "Centered Text";
  const Color SCREEN_BACKGROUND = RAYWHITE;

  InitWindow(SCREEN_WIDTH, SCREEN_HEIGHT, SCREEN_TITLE);

  float width = 400.0f;
  float height = 200.0f;
  Rectangle rect = (Rectangle){ (float)GetScreenWidth() / 2 - width / 2,
                                (float)GetScreenHeight() / 2 - height / 2,
                                width,
                                height };
  Color color = BLACK;
  float thickness = 5.0f;

  int num = 0;

  while (!WindowShouldClose()) {
    if (IsKeyPressed(KEY_UP)) {
      num++;
    }
    if (IsKeyPressed(KEY_DOWN)) {
      num--;
    }

    BeginDrawing();
    ClearBackground(SCREEN_BACKGROUND);

    DrawRectangleLinesEx(rect, thickness, color);
    center_and_draw_text(num, rect);

    EndDrawing();
  }

  CloseWindow();

  return 0;
}
